src/main/java/com/minecolonies/coremod/util/BlockPosUtil.java

```diff
@@ -127,6 +129,58 @@ public static BlockPos readFromByteBuf(@NotNull final ByteBuf buf)
     }
 
     /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    public static boolean isPositionSafe(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR
+                && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid()
+                && !sender.getEntityWorld().getBlockState(blockPos.up()).getMaterial().isLiquid();
+    }
+
+    /**
+     * this checks that you are not in the air or underground
+     * and if so it will look up and down for a good landing spot
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @return blockPos to be used for the TP
+     */
+    public static BlockPos findLand(final BlockPos blockPos, final World world)
+    {
+        int top = blockPos.getY();
+        int bot = 0;
+        int mid = blockPos.getY();
+
+        BlockPos foundland = null;
+        BlockPos tempPos = blockPos;
+        //We are doing a binary search to limit the amount of checks (usually at most 9 this way)
+        while (top >= bot)
+        {
+            tempPos = new BlockPos( tempPos.getX(),mid, tempPos.getZ());
+            Block blocks = world.getBlockState(tempPos).getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'blocks' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)